### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by xfangfang**
The design and implementation of this game are masterpieces, but it seems that there are some memory leaks. Please review the code, thanks~
btw, I'm working on porting this game to PSV, because the PSV has relatively small memory and video memory, which is why I found this problem.

### CODE REVIEW

File: src/types.c [hunk 1]
```diff
@@ -50,6 +50,7 @@ bool initText(Text* self, const char* str, SDL_Color color) {
       printf("Unable to create texture from rendered text! SDL Error: %s\n",
              SDL_GetError());
     } else {
+      if (self->origin) SDL_DestroyTexture(self->origin);
```
Comment (yujqiao) [inline src/types.c:53]: Any specific reason for adding this line and the line in `createText`? Will lacking of these lines cause a memory leak? A restriction I want to maintain is that `initX` should be called on uninitialized memory and `createX` should malloc an area of uninitialized memory and then initialize it.

**Comment by yujqiao** [comment]
Thanks for your interest and contribution! I have added a comment to your code.

Also, porting this game to PSV sounds very cool. Feel free to reach me if you have any difficulty. A major challenge of poring that comes to my mind is that the game doesn't support different resolutions and was written as if everyone has a large enough screen. This is one item in the to-do list but I don't know when I can find time to resolve that.

File: src/types.c (hunk 1 above)
Comment (xfangfang) [reply src/types.c:53]: there is a function named `setText` at types.c, it called `initText` every time the text changed.
In this case, the texture is repeatedly created when `initText` is called, resulting in a memory leak.

**Comment by xfangfang** [comment]
the first version is already done, you can check it at [github](https://github.com/xfangfang/DungeonRush-PSV) or [vitaDB](https://vitadb.rinnegatamante.it/#/info/578)(a psv homebrew store)

File: src/types.c (hunk 1 above)
Comment (yujqiao) [reply src/types.c:53]: You're right. Then can you move the line to `setText` thus leaving `createText` and `initText` unchanged? `initText` shouldn't assume the memory area has any valid address. The `if` statement breaks that.

Comment (xfangfang) [reply src/types.c:53]: I don't think that's a good idea. It can avoid possible memory leaks in the future. But if you insist, I can change it.

**Comment by yujqiao** [comment]
initX isn't always in pair with createX. It's possible that initX get a uninitialized memory and self-&gt;origin in that random area is not NULL. Then the program will destroy texture at accroding to a random adress.



---Original---

**Comment by yujqiao** [comment]
Just thought of firstly destroyText and then createText. I dont know if I have implemented destroyText but i think it's the way to go



---Original---

File: src/types.c (hunk 1 above)
Comment (xfangfang) [reply src/types.c:53]: Start a new PR.

**Comment by yujqiao** [comment]
Close by #21

**Pull request closed without merging**